### PR TITLE
BSP SPI Documentation

### PR DESCRIPTION
**Pull request by sugamarora99**

### CODE REVIEW

File: Docs/source/Software/BSP.rst [hunk 1]
```diff
@@ -135,3 +135,18 @@ Usage
...
     entire BPS takes, it is worth noting.
+
+SPI BSP: Clark Poon, Sijin Woo, and Sugam Arora
+===============================================
+
+Purpose
+    SPI is used for communication between the uC and the minion LTC6811s in the system.
+
+Usage
+    Our car uses the ``SPI1`` and ``SPI3`` busses. When calling the init function, you must specify which
```
Comment (manthanand) [inline Docs/source/Software/BSP.rst:146]: Include what each one is used for.

File: Docs/source/Software/BSP.rst [hunk 2]
```diff
@@ -135,3 +135,18 @@ Usage
...
+    SPI is used for communication between the uC and the minion LTC6811s in the system.
+
+Usage
+    Our car uses the ``SPI1`` and ``SPI3`` busses. When calling the init function, you must specify which 
+    port you intend to use (via the ``spi_port_t`` struct). The user can read/write on whichever bus they 
+    have chosen. 
+    This BSP includes IRQ Handlers for both buses that post the ``MinionsASIC_Mutex``.
+
+Additional Considerations
+    The ``SPI1`` and ``SPI3`` IRQs are only included in the RTOS version of the code.
```
Comment (manthanand) [inline Docs/source/Software/BSP.rst:152]: If there is any reason they should be added to the Bare-Metal version of the code I would include that.

Comment (sugamarora99) [reply Docs/source/Software/BSP.rst:153]: Last spring the team decided to leave these handlers out of the bare metal version

File: Docs/source/Software/BSP.rst [hunk 3]
```diff
@@ -135,3 +135,19 @@ Usage
...
+SPI BSP: Clark Poon, Sijin Woo, and Sugam Arora
+===============================================
+
+Purpose
+    SPI is used for communication between the uC and the minion LTC6811s in the system.
+
+Usage
+    Our car uses the ``SPI1`` and ``SPI3`` busses. The ``SPI1`` bus is for communication with the LTC6811s
+    and the ``SPI3`` bus is for communication with the AS8510 current sensor. When calling the init function, 
+    you must specify which port you intend to use (via the ``spi_port_t`` struct). The user can read/write on whichever bus they
```
Comment (manthanand) [inline Docs/source/Software/BSP.rst:148]: If there are #define for BareMetal and RTOS, I would explain why they are there and how to use them for each version

File: Docs/source/Software/BSP.rst [hunk 4]
```diff
@@ -135,3 +135,23 @@ Usage
...
+
+Usage
+    Our car uses the ``SPI1`` and ``SPI3`` busses. The ``SPI1`` bus is for communication with the LTC6811s
+    and the ``SPI3`` bus is for communication with the AS8510 current sensor. When calling the init function, 
+    you must specify which port you intend to use (via the ``spi_port_t`` struct). The user can read/write on whichever bus they 
+    have chosen. 
+    This BSP includes IRQ Handlers for both buses that post the ``MinionsASIC_Mutex``.
+
+    ``SPI_Wait()`` has two different versions - one for the bare-metal BPS code and one for the RTOS code. 
+    To use one of the versions, the user must have a line containing ``#define BAREMETAL`` or ``#define RTOS``
```
Comment (manthanand) [inline Docs/source/Software/BSP.rst:153]: The code is automatically compiled with #define RTOS. If they want it with bare metal, they just have to compile it with the BareMetal parameter.

File: Docs/source/Software/BSP.rst [hunk 5]
```diff
@@ -135,3 +135,23 @@ Usage
 Additional Considerations
     Increasing the clock speed does take more power. Although this is minor considering how much the
     entire BPS takes, it is worth noting.
+
+SPI BSP: Clark Poon, Sijin Woo, and Sugam Arora
+===============================================
+
+Purpose
+    SPI is used for communication between the uC and the minion LTC6811s in the system.
```
Comment (manthanand) [inline Docs/source/Software/BSP.rst:143]: Also add the AS8510 in this line.

File: Docs/source/Software/BSP.rst [hunk 6]
```diff
@@ -135,3 +135,23 @@ Usage
...
+    you must specify which port you intend to use (via the ``spi_port_t`` struct). The user can read/write on whichever bus they 
+    have chosen. 
+    This BSP includes IRQ Handlers for both buses that post the ``MinionsASIC_Mutex``.
+
+    ``SPI_Wait()`` has two different versions - one for the bare-metal BPS code and one for the RTOS code. 
+    To use one of the versions, the user must have a line containing ``#define BAREMETAL`` or ``#define RTOS``
+    in their code.
+
+Additional Considerations
+    The ``SPI1`` and ``SPI3`` IRQs are only included in the RTOS version of the code.
```
Comment (manthanand) [inline Docs/source/Software/BSP.rst:157]: Also include that if additional SPI buses are needed for other drivers, they just have to edit the spi_port_t struct typed variable.

Comment (manthanand) [reply Docs/source/Software/BSP.rst:157]: More information is probably included in Chase's comments in the file.

**Review by manthanand — APPROVED** [review]

**Review by manthanand — DISMISSED** [review]

**Review by ClarkPoon — DISMISSED** [review]

**Review by ClarkPoon — APPROVED** [review]

**Pull request merged**